server/subscribe.go

```diff
@@ -126,74 +181,22 @@ func (p *Plugin) getChannelsSubscribed(wh *webhook) (StringSet, error) {
 		return nil, err
 	}
 
-	subIds := subs.Channel.ById
-	instType := ji.GetType()
-
+	// Jira Cloud comment event. We need to fetch issue data because it is not expanded in webhook payload.
 	issue := &jwh.Issue
-	webhookEvents := wh.Events()
 	isCommentEvent := jwh.WebhookEvent == "comment_created" || jwh.WebhookEvent == "comment_updated" || jwh.WebhookEvent == "comment_deleted"
-
-	if isCommentEvent && instType == "cloud" {
-		// Jira Cloud comment event. We need to fetch issue data because it is not expanded in webhook payload.
+	if isCommentEvent && ji.GetType() == "cloud" {
 		issue, err = p.getIssueDataForCloudWebhook(ji, issue.ID)
```

Comment:
This `issue` var needs to be passed into `p.matchesSubscriptionFilters`, so it will have the detailed one when it needs to be fetched here.

Reply:
@mickmister I modified the webhook in place to fix it. However in what situations do we need to do this? Do I need to move this code upstream so it's covered for workflows as well?

Reply:
>  However in what situations do we need to do this?

The comment added/updated/deleted events for Jira Cloud do not supply the whole issue in the payload, so we have to fetch it.

> Do I need to move this code upstream so it's covered for workflows as well?

Yes. Maybe have a `fetchFullIssueIfNecessary` function or something, that has the `if` condition?

Reply:
0/5 `fetchFullIssueIfNecessary` would be very Java.

Reply:
@crspeller I can approve after moving the `getIssueDataForCloudWebhook` call out of the function

Reply:
@mickmister Latest commit does this.